File: src/services/admissionService.js
import api from "./api";

export const getAllAdmissions = () => {
  return api.get("/api/admissions");
};

export const saveAdmission = (admission) => {
  return api.post("/api/admissions", admission);
};

export const deleteAdmission = (id) => {
  return api.delete(`/api/admissions/${id}`);
};

// ✅ ADD THIS
export const updateAdmission = (id, admission) => {
  return api.put(`/api/admissions/${id}`, admission);
};

File: src/main/java/com/hospital/management/admission/service/AdmissionService.java
package com.hospital.management.admission.service;

import com.hospital.management.admission.dto.AdmissionRequestDTO;
import com.hospital.management.admission.dto.AdmissionResponseDTO;

import java.util.List;

public interface AdmissionService {

    AdmissionResponseDTO saveAdmission(AdmissionRequestDTO dto);

    List<AdmissionResponseDTO> getAllAdmissions();

    AdmissionResponseDTO getAdmissionById(Long id);

    AdmissionResponseDTO updateAdmission(Long id, AdmissionRequestDTO dto);

    void deleteAdmission(Long id);
}